pkg/types/aws_resource_descriptor.go

```diff
@@ -42,6 +43,13 @@ type AWSResourceDescriptor interface {
 	// should be the same. In other words, the Diff() method should be called
 	// with the same concrete implementing AWSResource type
 	Diff(AWSResource, AWSResource) string
+
+	// DiffReport returns a Reporter which provides the difference between two supplied
+	// AWSResources. The underlying types of the two supplied AWSResources should
+	// be the same. In other words, the DiffReport() method should be called with the
+	// same concrete implementing AWSResource type
+	DiffReport(AWSResource, AWSResource) *ackcompare.Reporter
```

Comment:
No need to add another `DiffReport` method to this interface. Simply go ahead and change the existing `Diff` method to return an `*ackcompare.Reporter` instead of a string.

Then, on line 165-171 of reconciler.go:

```go
		diff := r.rd.Diff(desired, latest)
		r.log.V(1).Info(
			"desired resource state has changed",
			"diff", diff,
			"arn", latest.Identifiers().ARN(),
			"is_adopted", isAdopted,
		)
```

simply change it to this:

```go
		diff := r.rd.Diff(desired, latest)
		r.log.V(1).Info(
			"desired resource state has changed",
			"diff", diff.String(),
			"arn", latest.Identifiers().ARN(),
			"is_adopted", isAdopted,
		)
```

Reply:
Code updated.